common/interfaces/user_interface/ui_scroll_page.c

```diff
@@ -0,0 +1,572 @@
+    (gp_scrollabe_page_lvgl->ui_arrow_pressed_style).body.grad_color = LV_COLOR_BLACK;
+    (gp_scrollabe_page_lvgl->ui_arrow_pressed_style).body.radius     = 30;
+    (gp_scrollabe_page_lvgl->ui_arrow_pressed_style).text.color      = LV_COLOR_WHITE;
+    lv_label_set_body_draw(gp_scrollabe_page_lvgl->p_ui_left_arrow_lvgl, true);
+    lv_label_set_body_draw(gp_scrollabe_page_lvgl->p_ui_right_arrow_lvgl, true);
+
+    /** 
+     * Calculate the number of pages/max number of times full scrolling can take place
+     * Total number of scrolls = Total height of content on page / Height of page
+     * We have already padded the content on page with label paddingLabel and therefore
+     * we do not need to consider padding here
+     */
+    int16_t totalPageHeight =
+        (int16_t)lv_page_get_scrl_height(gp_scrollabe_page_lvgl->p_ui_page_lvgl);
+    int16_t currPageHeight = (int16_t)lv_obj_get_height(gp_scrollabe_page_lvgl->p_ui_page_lvgl);
+    ASSERT(0 != currPageHeight);
+    gp_scrollabe_page_data->curr_page_num  = 1;
+    gp_scrollabe_page_data->total_page_num = totalPageHeight / currPageHeight;
+
+    /**
+     * Create buttons on the screen for cancellation and confirmation.
+     * These buttons will be visible conditionally (if the current page is the last page)
+     */
+    gp_scrollabe_page_lvgl->p_ui_cancel_btn_lvgl = lv_btn_create(lv_scr_act(), NULL);
+    ui_cancel_btn(gp_scrollabe_page_lvgl->p_ui_cancel_btn_lvgl, page_cancel_handler,
+                  gp_scrollabe_page_data->bool_accept_cancel_hidden);
+    lv_obj_set_size(gp_scrollabe_page_lvgl->p_ui_cancel_btn_lvgl, 16, 16);
+
+    gp_scrollabe_page_lvgl->p_ui_accept_btn_lvgl = lv_btn_create(lv_scr_act(), NULL);
+    ui_next_btn(gp_scrollabe_page_lvgl->p_ui_accept_btn_lvgl, page_accept_handler,
+                gp_scrollabe_page_data->bool_accept_cancel_hidden);
+    lv_obj_set_size(gp_scrollabe_page_lvgl->p_ui_accept_btn_lvgl, 16, 16);
+
+    /**
+     * Create a label gp_scrollabe_page_lvgl->p_ui_footnote_lvgl which holds the text that goes
+     * as part of a footnote on the current screen
+     */
+    gp_scrollabe_page_lvgl->p_ui_footnote_lvgl = lv_label_create(lv_scr_act(), NULL);
+
+    /* Update all icons: Left/right arrows, Accept/Cancel buttons and Footnote */
+    page_update_icons();
+
+    return;
+}
+
+void ui_scrollable_page(const char *p_page_ui_heading,
+                        const char *p_page_ui_body,
+                        e_scrollable_page_orientation_t page_orientation,
+                        bool bool_cancel_accept_btn_visible) {
+    if ((NULL == p_page_ui_heading) || (NULL == p_page_ui_body)) {
+        return;
+    }
+
+    gp_scrollabe_page_data = (scrolling_page_data_t *)malloc(sizeof(scrolling_page_data_t));
+    ASSERT(NULL != gp_scrollabe_page_data);
+
+    gp_scrollabe_page_data->p_ui_heading               = p_page_ui_heading;
+    gp_scrollabe_page_data->p_ui_body                  = p_page_ui_body;

```

Comment:
In most UI handlers, data is copied rather than passed as a pointer.
Considering there could be cases where the string is not NULL terminated, there should be a maximum size limit to avoid memory leaks.
Have we considered a max text size?

Reply:
Valid point, but the caller of this api must ensure null termination in string. Same is mentioned in the doxygen comments of the function.

Reply:
That works.